BodoSQL/calcite_sql/bodosql-calcite-application/src/main/java/org/apache/calcite/sql/validate/implicit/BodoTypeCoercionImpl.java

```diff
@@ -429,4 +431,102 @@ public boolean coalesceCoercionImpl(SqlCallBinding callBinding) {
     }
     return coerced;
   }
+
+  /**
+   * Extension of getWiderTypeForTwo to fix a bug in Array types.
+   */
+  @Override public @Nullable RelDataType getWiderTypeForTwo(
```

Comment:
These two functions were overloading the Calcite builtin. I moved them to our extension, which is the correct interface.